React storybook: issue importing from other libraries

## Expected Behavior

When used with storybook it should be possible to import from other internal libraries.

## Current Behavior

Storybook doesn't allow importing from another internal library or allow code that imports from another library. 

## Failure Information (for bugs)

Library A has storybook configured. This will work fine if the library has no dependencies on another internal library, however it will fail if another internal library is referenced either in the story or code that the story references. 

### Steps to Reproduce

1. Create a react library
2. Add storybook configuration to it
3. Add a dependency on another internal lib

### Context

Please provide any relevant information about your setup:

```
  @nrwl/angular : Not Found
  @nrwl/cli : 8.11.1
  @nrwl/cypress : 8.11.1
  @nrwl/eslint-plugin-nx : 8.11.1
  @nrwl/express : Not Found
  @nrwl/jest : 8.11.1
  @nrwl/linter : 8.11.1
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : 8.11.1
  @nrwl/schematics : Not Found
  @nrwl/tao : 8.11.1
  @nrwl/web : 8.11.1
  @nrwl/workspace : 8.11.1
  typescript : 3.5.3
```

### Failure Logs
Output from storybook:
```

ERROR in ./libs/core/src/lib/core.tsx
Module not found: Error: Can't resolve '@storybook-issue/hello-world' in 'User\Documents\Github\storybook-issue\libs\core\src\lib'
 @ ./libs/core/src/lib/core.tsx 3:0-58 10:43-53
 @ ./libs/core/src/lib/core.stories.tsx
 @ ./libs/core/src/lib sync \.stories\.tsx?$
 @ ./libs/core/.storybook/config.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./libs/core/.storybook/config.js ./node_modules/webpack-hot-middleware/client.js?reload=true&quiet=true
```

## Other

I've recreated the bug here: https://github.com/allforabit/storybook-issue

--- a/e2e/storybook/src/storybook.test.ts
+++ b/e2e/storybook/src/storybook.test.ts
@@ -143,6 +143,173 @@ forEachCli(() => {
         );
       }, 1000000);
     });
+
+    describe('build storybook', () => {
+      it('should build an Angular based storybook', () => {
+        ensureProject();
+
+        const angularStorybookLib = uniq('test-ui-lib');
+        createTestUILib(angularStorybookLib);
+        runCLI(
+          `generate @nrwl/angular:storybook-configuration ${angularStorybookLib} --generateStories --no-interactive`
+        );
+
+        // build Angular lib
+        runCLI(`run ${angularStorybookLib}:build-storybook`);
+        checkFilesExist(`dist/storybook/${angularStorybookLib}/index.html`);
+        expect(
+          readFile(`dist/storybook/${angularStorybookLib}/index.html`)
+        ).toContain(`<title>Storybook</title>`);
+      }, 1000000);
+
+      it('should build a React based storybook', () => {
+        ensureProject();
+
+        const reactStorybookLib = uniq('test-ui-lib-react');
+        runCLI(
+          `generate @nrwl/react:lib ${reactStorybookLib} --no-interactive`
+        );
+        runCLI(
+          `generate @nrwl/react:storybook-configuration ${reactStorybookLib} --generateStories --no-interactive`
+        );
+
+        // build React lib
+        runCLI(`run ${reactStorybookLib}:build-storybook`);
+        checkFilesExist(`dist/storybook/${reactStorybookLib}/index.html`);
+        expect(
+          readFile(`dist/storybook/${reactStorybookLib}/index.html`)
+        ).toContain(`<title>Storybook</title>`);
+      }, 1000000);
+
+      it('should build a React based storybook that references another lib', () => {
+        ensureProject();
+
+        const reactStorybookLib = uniq('test-ui-lib-react');
+        runCLI(
+          `generate @nrwl/react:lib ${reactStorybookLib} --no-interactive`
+        );
+        runCLI(
+          `generate @nrwl/react:storybook-configuration ${reactStorybookLib} --generateStories --no-interactive`
+        );
+
+        const anotherReactLib = uniq('test-another-lib-react');
+        runCLI(`generate @nrwl/react:lib ${anotherReactLib} --no-interactive`);
+        // create a React component we can reference
+        writeFileSync(
+          tmpProjPath(`libs/${anotherReactLib}/src/lib/mytestcmp.tsx`),
+          `
+            import React from 'react';
+            
+            /* eslint-disable-next-line */
+            export interface MyTestCmpProps {}
+            
+            export const MyTestCmp = (props: MyTestCmpProps) => {
+              return (
+                <div>
+                  <h1>Welcome to test cmp!</h1>
+                </div>
+              );
+            };
+            
+            export default MyTestCmp;
+        `
+        );
+        // update index.ts and export it
+        writeFileSync(
+          tmpProjPath(`libs/${anotherReactLib}/src/index.ts`),
+          `
+            export * from './lib/mytestcmp';
+        `
+        );
+
+        // create a story in the first lib to reference the cmp from the 2nd lib
+        writeFileSync(
+          tmpProjPath(
+            `libs/${reactStorybookLib}/src/lib/myteststory.stories.tsx`
+          ),
+          `
+            import React from 'react';
+            
+            import { MyTestCmp, MyTestCmpProps } from '@proj/${anotherReactLib}';
+    
+            export default {
+              component: MyTestCmp,
+              title: 'MyTestCmp',
+            };
+    
+            export const primary = () => {
+              /* eslint-disable-next-line */
+              const props: MyTestCmpProps = {};
+    
+              return <MyTestCmp />;
+            };
+        `
+        );
+
+        // build React lib
+        runCLI(`run ${reactStorybookLib}:build-storybook`);
+        checkFilesExist(`dist/storybook/${reactStorybookLib}/index.html`);
+        expect(
+          readFile(`dist/storybook/${reactStorybookLib}/index.html`)
+        ).toContain(`<title>Storybook</title>`);
+      }, 1000000);
+
+      it('should build an Angular based storybook that references another lib', () => {
+        ensureProject();
+
+        const angularStorybookLib = uniq('test-ui-lib');
+        createTestUILib(angularStorybookLib);
+        runCLI(
+          `generate @nrwl/angular:storybook-configuration ${angularStorybookLib} --generateStories --no-interactive`
+        );
+
+        // create another lib with a component
+        const anotherTestLib = uniq('test-another-lib');
+        runCLI(`g @nrwl/angular:library ${anotherTestLib} --no-interactive`);
+        runCLI(
+          `g @nrwl/angular:component my-test-cmp --project=${anotherTestLib} --no-interactive`
+        );
+
+        // update index.ts and export it
+        writeFileSync(
+          tmpProjPath(`libs/${anotherTestLib}/src/index.ts`),
+          `
+            export * from './lib/my-test-cmp/my-test-cmp.component';
+        `
+        );
+
+        // create a story in the first lib to reference the cmp from the 2nd lib
+        writeFileSync(
+          tmpProjPath(
+            `libs/${angularStorybookLib}/src/lib/myteststory.stories.ts`
+          ),
+          `
+            import { MyTestCmpComponent } from '@proj/${anotherTestLib}';
+
+            export default {
+              title: 'My Test Cmp',
+            };
+            
+            let x = 'hi';
+            
+            export const primary = () => ({
+              moduleMetadata: {
+                imports: [],
+              },
+              component: MyTestCmpComponent,
+              props: {},
+            });
+        `
+        );
+
+        // build Angular lib
+        runCLI(`run ${angularStorybookLib}:build-storybook`);
+        checkFilesExist(`dist/storybook/${angularStorybookLib}/index.html`);
+        expect(
+          readFile(`dist/storybook/${angularStorybookLib}/index.html`)
+        ).toContain(`<title>Storybook</title>`);
+      }, 1000000);
+    });
   });
 });
 

--- a/packages/storybook/src/schematics/configuration/lib-files/.storybook/webpack.config.js__tmpl__
+++ b/packages/storybook/src/schematics/configuration/lib-files/.storybook/webpack.config.js__tmpl__
@@ -1,10 +1,21 @@
+const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
 const rootWebpackConfig = require('<%= offsetFromRoot %>../.storybook/webpack.config');
 // Export a function. Accept the base config as the only param.
 module.exports = async ({ config, mode }) => {
   config = await rootWebpackConfig({ config, mode });
+
+  const tsPaths = new TsconfigPathsPlugin({
+    configFile: './tsconfig.base.json',
+   });
+
+  config.resolve.plugins
+    ? config.resolve.plugins.push(tsPaths)
+    : (config.resolve.plugins = [tsPaths])
+
   <% if(uiFramework === '@storybook/react') { %>
   config.resolve.extensions.push('.tsx');
   config.resolve.extensions.push('.ts');
+    
   config.module.rules.push({
     test: /\.(ts|tsx)$/,
     loader: require.resolve('babel-loader'),

